tests/integration/backward_compatible/db_test.py

```diff
@@ -0,0 +1,123 @@
+        cls.client.shutdown()
+        cls.rc.terminateCluster(cls.cluster.id)
+        cls.rc.exit()
+
+    def _create_mapping(self, value_format="INTEGER"):
+        if not self.is_v5_or_newer_server:
+            # Implicit mappings are removed in 5.0
+            return
+        q = f"""
+        CREATE MAPPING "{self.map_name}" (
+            __key INT,
+            this {value_format}
+        )
+        TYPE IMaP 
+        OPTIONS (
+            'keyFormat' = 'int',
+            'valueFormat' = '{value_format.lower()}'
+        )
+        """
+        c = self.conn.cursor()
+        c.execute(q)
+
+    def _populate_map(self, entry_count=10, value_factory=lambda v: v):
+        entries = [(i, value_factory(i)) for i in range(entry_count)]
+        c = self.conn.cursor()
+        c.executemany(f'INSERT INTO "{self.map_name}" VALUES(?, ?)', entries)
+
+
+class DbapiTest(DbapiTestBase):
+    def test_fetchone(self):
+        self._create_mapping()
+        entry_count = 11
+        self._populate_map(entry_count)
+        c = self.conn.cursor()
+        c.execute(f'SELECT * FROM "{self.map_name}" where __key < ? order by __key', (5,))
+        self.assertEqual(0, c.rownumber)
+        row = c.fetchone()
+        self.assertEqual((0, 0), (row.get_object("__key"), row.get_object("this")))
+        self.assertEqual(1, c.rownumber)
+        row = c.fetchone()
+        self.assertEqual((1, 1), (row.get_object("__key"), row.get_object("this")))
+        self.assertEqual(2, c.rownumber)
+
+    def test_fetchmany(self):
+        self._create_mapping()
+        entry_count = 11
+        self._populate_map(entry_count)
+        c = self.conn.cursor()
+        c.execute(f'SELECT * FROM "{self.map_name}" where __key < ? order by __key', (5,))
+        self.assertEqual(0, c.rownumber)
+        result = list(c.fetchmany(3))
+        self.assertCountEqual(
+            [(i, i) for i in range(3)],
+            [(row.get_object("__key"), row.get_object("this")) for row in result],
+        )
+        self.assertEqual(3, c.rownumber)
+        result = list(c.fetchmany(3))
+        self.assertCountEqual(
+            [(i, i) for i in range(3, 5)],
+            [(row.get_object("__key"), row.get_object("this")) for row in result],
+        )
+        self.assertEqual(5, c.rownumber)
+
+    def test_fetchall(self):
+        self._create_mapping()
+        entry_count = 11
+        self._populate_map(entry_count)
+        c = self.conn.cursor()
+        c.execute(f'SELECT * FROM "{self.map_name}" where __key < ? order by __key', (5,))
+        self.assertEqual(0, c.rownumber)
+        result = list(c.fetchall())
+        self.assertCountEqual(
+            [(i, i) for i in range(5)],
+            [(row.get_object("__key"), row.get_object("this")) for row in result],
+        )
+        self.assertEqual(5, c.rownumber)
+
+    def test_cursor_connection(self):
+        c = self.conn.cursor()
+        self.assertEqual(self.conn, c.connection)
```

Comment:
can you add a test for description, as we return our own class there?

Reply:
`DescriptionColumn` is not a class but a named tuple, so equivalent to the tuple in the spec. but will add a test.